cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1129,22 +1129,33 @@ private boolean isKafkaClusterReady(Kafka kafka) {
                 && kafka.getStatus().getConditions().stream().anyMatch(condition -> condition.getType().equals("Ready") && condition.getStatus().equals("True"));
     }
 
+    private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> requestRebalance(Reconciliation reconciliation,
+                                                                                   String host, CruiseControlApi apiClient, KafkaRebalance kafkaRebalance,
+                                                                                   boolean dryrun, AbstractRebalanceOptions.AbstractRebalanceOptionsBuilder<?, ?> rebalanceOptionsBuilder,
+                                                                                   boolean stopOngoingExecution) {
+        return requestRebalance(reconciliation, host, apiClient, kafkaRebalance, dryrun, rebalanceOptionsBuilder, null, stopOngoingExecution);
+    }
+
     private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> requestRebalance(Reconciliation reconciliation,
                                                           String host, CruiseControlApi apiClient, KafkaRebalance kafkaRebalance,
                                                           boolean dryrun, AbstractRebalanceOptions.AbstractRebalanceOptionsBuilder<?, ?> rebalanceOptionsBuilder) {
-        return requestRebalance(reconciliation, host, apiClient, kafkaRebalance, dryrun, rebalanceOptionsBuilder, null);
+        return requestRebalance(reconciliation, host, apiClient, kafkaRebalance, dryrun, rebalanceOptionsBuilder, null, false);
     }
 
 
     private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> requestRebalance(Reconciliation reconciliation, String host, CruiseControlApi apiClient, KafkaRebalance kafkaRebalance,
```

Comment:
I noticed that this method is actually called only from the previous one by providing `userTaskID` as null (so in general it means we are passing `userTaskID` null everywhere). Taking this out, anyway I think that adding one more method is confusing. Can we conflate in just one method with both `userTaskID` parameter and `stopOngoingExecution` and using it? (of course `userTaskID` will be always null as today, we can figure out why later).

Reply:
You are right, adding another method doesn't make sense. I think previously we used to get rebalance status via rebalance endpoint therefore needed to specify user task id. After https://github.com/strimzi/strimzi-kafka-operator/pull/10607, we no longer need it. So I think we should stop passing it to the method.

Reply:
Perhaps, that should be done in a separate PR?

Reply:
> Perhaps, that should be done in a separate PR?

yeah I agree and we'll think more through. For now I am happy not having an additional one :-)